pkg/logql/engine.go

```diff
@@ -99,17 +106,35 @@ type query struct {
 }
 
 // Exec Implements `Query`
-func (q *query) Exec(ctx context.Context) (promql.Value, error) {
+func (q *query) Exec(ctx context.Context) (Result, error) {
+	log, ctx := spanlogger.New(ctx, "Engine.Exec")
```

Comment:
Part of me thinks this should be `query.Exec`, but I think it's also nicer to have `Engine.Exec` in traces.